### PR TITLE
[Spot] Disallow spot autostop

### LINKED ISSUE #1472: [Spot] Inconsistency between sky stop and sky autostop
Found this during https://github.com/skypilot-org/skypilot/pull/1470, we do not support `sky stop` a spot VM but allow `sky autostop` it, which should be banned as well because its disk is not persistent on AWS?

```
(sky-tmp) weichiang@mbp sky % sky stop sky-2edf-weichiang
Stopping 1 cluster: sky-2edf-weichiang. Proceed? [Y/n]:
Stopping cluster 'sky-2edf-weichiang'... skipped.
  Stopping spot instances is not supported as the attached disks will be lost.
  To terminate the cluster instead, run: sky down sky-2edf-weichiang
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--


(sky-tmp) weichiang@mbp sky % sky autostop sky-2edf-weichiang
Scheduling autostop on 1 cluster: sky-2edf-weichiang. Proceed? [Y/n]:
Scheduling autostop on cluster 'sky-2edf-weichiang'...done
  The cluster will be autostopped after 5 minutes of idleness.
  To cancel the autostop, run: sky autostop sky-2edf-weichiang --cancel
Scheduling autostop on 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
```

### PR DESCRIPTION
**Pull request by ewzeng**
This PR fixes #1472 by disallowing spot autostop.

Tested:

- [x] `sky launch --cloud gcp --use-spot -i 5` (disallow)
- [x] `sky launch --cloud gcp --use-spot --down -i 5`
- [x] `sky autostop [spot cluster]` (disallow)
- [x] `sky autostop [spot cluster] -i 5` (disallow)
- [x] `sky autostop [spot cluster] --cancel` 
- [x] `sky autostop [spot cluster] --down`

### CODE REVIEW

File: sky/execution.py [hunk 1]
```diff
@@ -210,6 +210,11 @@ def _execute(
             if not down:
                 requested_features.add(
                     clouds.CloudImplementationFeatures.AUTOSTOP)
+                # TODO(suquark): enable GCP+spot to be stopped in the future.
```
Comment (Michaelvll) [inline sky/execution.py:213]: Seems not needed?

File: sky/core.py [hunk 2]
```diff
@@ -429,6 +429,14 @@ def autostop(
         raise exceptions.NotSupportedError(
             f'{operation} cluster {cluster_name!r} with backend '
             f'{backend.__class__.__name__!r} is not supported.')
+    elif handle.launched_resources.use_spot and not down and not is_cancel:
+        # Disable spot instances to be stopped.
```
Comment (Michaelvll) [inline sky/core.py:433]: ... to be autostopped?

**Review by Michaelvll — APPROVED** [review]
Thanks for the quick fix @ewzeng! LGTM.

File: sky/execution.py (hunk 1 above)
Comment (ewzeng) [reply sky/execution.py:213]: I was under the impression that if we support GCP+spot stop, then we would support `sky launch --cloud gcp --use-spot -i 5`?

Comment (Michaelvll) [reply sky/execution.py:213]: Yes, then should we change it to `TODO: allow autostop for spot when stopping is supported.`

Comment (ewzeng) [reply sky/execution.py:213]: Ah, got it. That makes sense; I will fix

Comment (ewzeng) [reply sky/execution.py]: Updated!

File: sky/core.py (hunk 2 above)
Comment (ewzeng) [reply sky/core.py]: Updated!

**Pull request merged**